frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts

```diff
@@ -133,11 +155,34 @@ export class WorkPackageTimelineTableController extends UntilDestroyedMixin impl
       takeUntil(this.querySpace.stopAllSubscriptions),
       filter(() => this.initialized && this.wpTableTimeline.isVisible)
     )
-      .subscribe(([orderedRows, changes, timelineState]) => {
-        // Remember all visible rows in their order of appearance.
-        this.workPackageIdOrder = orderedRows.filter(row => !row.hidden);
-        this.refreshView();
-      });
+    .subscribe(([orderedRows, changes, timelineState]) => {
+      // Remember all visible rows in their order of appearance.
+      this.workPackageIdOrder = orderedRows.filter(row => !row.hidden);
+      this.orderedRows = orderedRows;
+      this.refreshView();
+    });
+
+    merge(
+      // Refresh the last collapsed/expanded group header cells when its collapsed state changes
+      this.querySpace.collapsedGroups.changes$().pipe(filter(collapsedGroupsChange => collapsedGroupsChange != null)),
+      // Refresh all the collapsed group header cells whenever the query changes
+      this.querySpace.initialized.values$().pipe(switchMap(() => this.querySpace.tableRendered.values$().pipe(take(1), map(() => false)))),
+    )
+    .pipe(
+      this.untilDestroyed(),
+      takeUntil(this.querySpace.stopAllSubscriptions),
+      filter(() => this.initialized && this.wpTableTimeline.isVisible),
+    )
+    .subscribe((change:{[identifier:string]:boolean} | false) => {
+      const collapsedGroupsChange = change || this.querySpace.collapsedGroups.value;
+      const refreshAllGroupHeaderCells = !change;
+
+      this.manageCollapsedGroupHeaderCells(this.querySpace.groups.value!,
+                                            collapsedGroupsChange,
+                                            this.querySpace.results.value!.elements,
+                                            this.collapsedGroupsCellsMap,
+                                            refreshAllGroupHeaderCells);
```

Comment:
Why pass these three parameters starting with `this.` if the called method is in the same class and as such will have access to the values directly?